aicsimageio/readers/lif_reader.py

```diff
@@ -0,0 +1,751 @@
+        S: int = 0,
+        **kwargs
+    ):
+        # Run super init to check filepath provided
+        super().__init__(data, **kwargs)
+
+        # Store parameters needed for _daread
+        self.chunk_by_dims = chunk_by_dims
+        self.specific_s_index = S
+        lif = LifFile(filename=self._file)
+        #  _chunk_offsets is a list of ndarrays (only way I could deal with inconsistent scene shape)
+        self._chunk_offsets, self._chunk_lengths = LifReader._compute_offsets(lif=lif)
+
+    @staticmethod
+    def _is_this_type(buffer: io.BytesIO) -> bool:
+        """
+        Test the file that was provided to check that the header is consistent with this reader.
+
+        Parameters
+        ----------
+        buffer: io.BytesIO
+            This is the contents of the file path the LifReader was initialized with.
+
+        Returns
+        -------
+        True / False
+            True if it has the right header byte structure False if it does not.
+
+        """
+        with BufferReader(buffer) as buffer_reader:
+            header = buffer_reader.read_bytes(n_bytes=8)
+
+            # if the buffer is to short return false
+            if len(buffer_reader.endianness) < 2 or len(header) < 8:
+                return False
+            # check for the magic byte
+            if buffer_reader.endianness[0] != LifReader.LIF_MAGIC_BYTE and header[1] != LifReader.LIF_MAGIC_BYTE:
+                return False
+            # check for the memory byte, if magic byte and memory byte are present return true
+            if header[6] == LifReader.LIF_MEMORY_BYTE:
+                return True
+        return False
+
+    @staticmethod
+    def _read_image(img: Path, offsets: List[np.ndarray], r_length: np.ndarray, read_dims: Optional[Dict[str, int]] = None) -> Tuple[np.ndarray, List[Tuple[str, int]]]:
+        """
+        Read and return the squeezed image data requested along with the dimension info that was read.
+
+        Parameters
+        ----------
+        img: Path
+            Path to the LIF file to read.
+        offsets: List[numpy.ndarray]
+            A List of numpy ndarrays offsets, see _compute_offsets for more details.
+        r_length: numpy.ndarray
+            A 1D numpy array of read lengths, the index is the scene index
+        read_dims: Optional[Dict[str, int]]
+            The dimensions to read from the file as a dictionary of string to integer.
+            Default: None (Read all data from the image)
+
+        Returns
+        -------
+        data: np.ndarray
+            The data read for the dimensions provided.
+        read_dimensions: List[Tuple[str, int]]]
+            The dimension sizes that were returned from the read.
+        """
+        # Catch optional read dim
+        if read_dims is None:
+            read_dims = {}
+
+        # Read image
+        log.debug(f"Reading dimensions: {read_dims}")
+        data, dims = LifReader._get_item_as_bitmap(img, offsets, r_length, read_dims)
```

Comment:
Please move `_get_item_as_bitmap` to above this function. _used_ function goes before _using_ function.